enterprise/server/remote_execution/filecache/filecache.go

```diff
@@ -90,12 +90,18 @@ func sizeFn(value interface{}) int64 {
 }
 
 func evictFn(value interface{}, reason lru.EvictionReason) {
-	if v, ok := value.(*entry); ok {
-		syscall.Unlink(v.value)
-		if reason == lru.SizeEviction {
-			age := time.Since(time.UnixMicro(v.addedAtUsec)).Microseconds()
-			metrics.FileCacheLastEvictionAgeUsec.Set(float64(age))
-		}
+	// TODO(sluongng): consider using Go generic to avoid type assertion.
+	v, ok := value.(*entry)
+	if !ok {
+		log.Warningf("could not cast filecache's value to entry struct: %v", value)
+		return
+	}
+
+	err := syscall.Unlink(v.value)
+	log.Errorf("failed to unlink filecache entry %q: %s", v.value, err)
```

Comment:
(missing `err != nil` check)

```suggestion
  if err := syscall.Unlink(v.value); err != nil {
	  log.Errorf("Failed to unlink filecache entry %q: %s", v.value, err)
  }
```